packages/fxa-auth-server/fxa-oauth-server/lib/grant.js

```diff
@@ -182,5 +187,20 @@ function generateIdToken(grant, access) {
     claims.acr = 'AAL' + grant.aal;
   }
 
-  return ID_TOKEN_KEY.sign(claims);
+  return sign(claims);
```

Comment:
(Commenting on this line, but it's about the calculation of `at_hash` a few lines above).

I'm pretty sure this calculation of `at_hash` is wrong, and will be even wronger when generating JWT access tokens :-(

Per https://openid.net/specs/openid-connect-core-1_0.html#CodeIDToken

> Its value is the base64url encoding of the left-most half of the hash of the octets of the ASCII representation
> of the access_token value, where the hash algorithm used is the hash algorithm used in the alg Header
> Parameter of the ID Token's JOSE Header. For instance, if the alg is RS256, hash the access_token value with
> SHA-256, then take the left-most 128 bits and base64url encode them. The at_hash value is a case sensitive
> string.

Since we expect clients to (1) validate this hash locally and (2) treat access tokens as opaque strings, the only sensible value to hash here is the ascii string `access_token` value as seen by the client.  For DB tokens that's the hex string, and for JWTs that's the full JWT.